packages/backend-common/src/service/lib/ServiceBuilderImpl.ts

```diff
@@ -120,34 +154,96 @@ export class ServiceBuilderImpl implements ServiceBuilder {
         reject(e);
       });
 
-      const server = stoppable(
-        app.listen(port, host, () => {
-          logger.info(`Listening on ${host}:${port}`);
-        }),
-        0,
-      );
+      let server: http.Server;
+
+      if (httpsSettings) {
+        logger.info('Initializing https server');
+
+        let credentials: { key: string; cert: string };
+        const signingOptions: any = httpsSettings?.certificate;
+
+        if (signingOptions?.algorithm !== undefined) {
+          const certificateAttributes: Array<any> = [];
+
+          Object.getOwnPropertyNames(signingOptions?.attributes).forEach(
+            propertyName => {
+              const propertyValue = (signingOptions?.attributes as any)[
+                propertyName
+              ];
+
+              certificateAttributes.push({
+                name: propertyName,
+                value: propertyValue,
+              });
+            },
+          );
+
+          // TODO: Create a type def for selfsigned.
+          const signatures = require('selfsigned').generate(
```

Comment:
I'd say it's worth logging before this step, something like "generating self-signed certificate with attributes ...". Just to provide a bit of information and show what's going on, as this step can take a while.